numba/targets/linalg.py

```diff
@@ -1691,3 +1691,319 @@ def det_impl(a):
         return sgn * np.exp(slogdet)
 
     return det_impl
+
+
+def _get_norm_impl(a, ord_flag):
+    # This function is quite involved as norm supports a large
+    # range of values to select different norm types via kwarg `ord`.
+    # The implementation below branches on dimension of the input
+    # (1D or 2D). The default for `ord` is `None` which requires
+    # special handling in numba, this is dealt with first in each of
+    # the dimension branches. Following this the various norms are
+    # computed via code that is in most cases simply a loop version
+    # of a ufunc based version as found in numpy.
+
+    # The following is common to both 1D and 2D cases.
+    # Convert typing floats to numpy floats for use in the impl.
+    # The return type is always a float, numba differs from numpy in
+    # that it returns an input precision specific value whereas numpy
+    # always returns np.float64.
+    nb_ret_type = getattr(a.dtype, "underlying_float", a.dtype)
+    np_ret_type = np_support.as_dtype(nb_ret_type)
+
+    np_dtype = np_support.as_dtype(a.dtype)
+
+    xxnrm2_sig = types.intc(types.char,  # kind
+                            types.intp,  # n
+                            types.CPointer(a.dtype),  # x
+                            types.intp,  # incx
+                            types.CPointer(nb_ret_type))
+
+    xxnrm2 = types.ExternalFunction("numba_xxnrm2", xxnrm2_sig)
+
+    kind = ord(get_blas_kind(a.dtype, "norm"))
+
+    one = 1
+
+    FORCE_CONTIG = a.layout not in 'CF'
+
+    if a.ndim == 1:
+        # 1D cases
+
+        # Used if order is None or order == 2 : see note below
+        @jit(nopython=True)
+        def oneD_none_or_2_impl(a, order=None):
+            # Just ignore order, calls are guarded to only come
+            # from cases where order=None or order=2.
+            n = len(a)
+            # Call L2-norm routine from BLAS
+            ret = np.empty((1,), dtype=np_ret_type)
+            jmp = int(a.strides[0] / a.itemsize)
+            r = xxnrm2(
```

Comment:
(you may split it into two different functions, `xsnrm2` and `xdnrm2`, if necessary)